content/token22-on-chain.md

```diff
@@ -0,0 +1,1483 @@
+---
+title: Supporting Token Extension Program in on-chain programs
+objectives:
+- Accept both token programs' accounts, and mints in your program
+- Explain the differences between the Token Program and Token Extension programs
+- Explain how to use Anchor Interfaces
+---
+
+# Summary
+
+- The `Token Extension Program` is a superset of the `Token Program` with a different program id
+- `token_program` is an Anchor account constraint allowing you to verify an account belongs to a specific token program
+- Anchor introduced the concept of Interfaces to easily allow for programs to support interaction with both `Token Program` and `Token Extension Program`
+
+# Overview
+
+The `Token Extension Program` is a program on Solana mainnet that provides additional functionality to Solana tokens and mints. The `Token Extension Program` is a superset of the `Token Program`. Essentially it is a byte for byte recreation with additional functionality tagged on at the end. However they are sill separate programs. With two types of Token Programs, we must anticipate being sent the program type in instructions.
+
+In this lesson, you'll learn how to design your program to accept `Token Program` and `Token Extension Program` accounts using Anchor. You will also learn how to interact with `Token Extension Program` accounts, identifying which token program an account belongs to, and some differences between `Token Program` and the `Token Extension Program` on-chain.
+
+## Difference between legacy Token Program and Token Extension Program
+
+We must clarify that the `Token Extension Program` is separate from the original `Token Program`. The `Token Extension Program` is a superset of the original `Token Program`, meaning all the instructions and functionality in the original `Token Program` come with the `Token Extension Program`.
```

Comment:
This repeats the information from the other chapter. If we have multiple explanation of the same concept, one will end up being better maintained than the other. Just use a deep link to some place where the differences between the token program and token extensions program are discussed.